### PR TITLE
Improve rendering of page title

### PR DESCRIPTION
**Pull request by dzhuang**
- [x]  add tests

### CODE REVIEW

**Comment by dzhuang** [comment]
@inducer What do you think of this?

File: course/page/base.py [hunk 1]
```diff
@@ -388,8 +389,9 @@ def __init__(self, vctx, location, page_desc):
             self.is_optional_page = getattr(page_desc, "is_optional_page", False)
 
         else:
-            from warnings import warn
-            warn(_("Not passing page_desc to PageBase.__init__ is deprecated"),
+            if vctx is not None:
+                vctx.add_warning(
+                    _("Not passing page_desc to PageBase.__init__ is deprecated"),
```
Comment (inducer) [inline course/page/base.py]: This doesn't have much to do with course content--I don't think this warning belongs in validation output.

File: course/page/base.py [hunk 2]
```diff
@@ -765,9 +767,11 @@ def __init__(self, vctx, location, page_desc):
...
                 md_body = self.markup_body_for_title()
             except NotImplementedError:
-                from warnings import warn
-                warn(_("PageBaseWithTitle subclass '%s' does not implement "
-                        "markdown_body_for_title()")
+                if vctx is not None:
+                    vctx.add_warning(
+                        location,
+                        _("PageBaseWithTitle subclass '%s' does not implement "
+                          "markdown_body_for_title()")
```
Comment (inducer) [inline course/page/base.py]: Same.

Comment (inducer) [reply course/page/base.py]: Also, that warning gets the method name wrong. (Not introduced by you, but worth fixing nonetheless.)

File: course/page/base.py [hunk 3]
```diff
@@ -780,6 +784,13 @@ def __init__(self, vctx, location, page_desc):
                         _("no title found in body or title attribute"))
                     % (location))
 
+        from markdown import markdown
+        from django.utils.html import strip_tags
+        title = strip_tags(markdown(title))
```
Comment (inducer) [inline course/page/base.py]: Assuming `strip_tags` removes all HTML? If so, that seems fine. The title *is* fished out of the Markdown after all... Though I still don't fully understand the failure mode, as I noted.

File: course/page/base.py (hunk 1 above)
Comment (dzhuang) [reply course/page/base.py]: You're right, Actually I'm now reverting that.

File: course/page/base.py (hunk 3 above)
Comment (dzhuang) [reply course/page/base.py]: Both `**important**` and  `<b>important</b>`  are expected to be `important` in title.

File: course/page/base.py (hunk 2 above)
Comment (dzhuang) [reply course/page/base.py]: > that warning gets the method name wrong

Will fix that.

**Comment by dzhuang** [comment]
Updated.

File: course/templates/course/flow-page.html [hunk 4]
```diff
@@ -285,7 +285,7 @@
               {% for other_page_data in all_page_data %}
               <li>
                 <a href="{% url "relate-view_flow_page" course.identifier flow_session.id other_page_data.page_ordinal %}">
-                  {{ other_page_data.human_readable_ordinal }}: {{ other_page_data.title }}
+                  {{ other_page_data.human_readable_ordinal }}: {{ other_page_data.title|safe|striptags }}
```
Comment (inducer) [inline course/templates/course/flow-page.html]: Why is it `|striptags` in some places and not in others?

Comment (dzhuang) [reply course/templates/course/flow-page.html]: Sorry, it's my bad, that should be removed.  I should have proof read before submit.

Comment (dzhuang) [reply course/templates/course/flow-page.html]: Updated.

**Comment by inducer** [comment]
LGTM

**Comment by dzhuang** [comment]
Thanks for reviewing this.

**Pull request merged**